### ISSUE TITLE
Pythonic native arrays

### ISSUE THREAD

**Issue by AlexDaniel** [rakudo/rakudo#2525]
Let's get the element on index `-1`:
```perl6
my @a = 42, 50, 60; my $x = 0; say @a[$x-1] # returns a Failure
```
```
Index out of range. Is: -1, should be in 0..^Inf
  in block <unit> at -e line 1
```

Now with native arrays:

```perl6
my int8 @a = 42, 50, 60; my $x = 0; say @a[$x-1] # returns the last element
```

```
60
```

Is this behavior right?

**Comment by jnthn** [comment]
No; we should complain about this.

**Comment by lizmat** [comment]
If we die, perhaps this should be done at VM level?  Or do we need that behaviour of nqp::atpos ?

**Comment by lizmat** [comment]
@AlexDaniel: why the "shaped" label?  On shaped native arrays, this already dies:

    my int8 @a[3] = 42, 50, 60; my $x = 0; say @a[$x-1]
    # Index -1 for dimension 1 out of range (must be 0..2)

**Comment by AlexDaniel** [comment]
@lizmat sorry, mislabeled.

**Comment by lizmat** [comment]
Fixed with dd2af90cca

**Comment by lizmat** [comment]
Test added with https://github.com/rakudo/rakudo/commit/af96fbb438
